packages/viem/scripts/chaingen.ts

```diff
@@ -0,0 +1,159 @@
+  return (
+    parts[0].toLowerCase() +
+    parts
+      .slice(1)
+      .map((part) => part[0].toUpperCase() + part.substring(1))
+      .join('')
+  )
+}
+
+async function nativeCurrency(
+  client: PublicClient,
+  address: Address | undefined,
+): Promise<NativeCurrency> {
+  if (!address) {
+    return { name: 'Ether', symbol: 'ETH', decimals: 18 }
+  }
+
+  const [name, symbol, decimals] = await Promise.all([
+    client.readContract({ address, abi: erc20Abi, functionName: 'name' }),
+    client.readContract({ address, abi: erc20Abi, functionName: 'symbol' }),
+    client.readContract({ address, abi: erc20Abi, functionName: 'decimals' }),
+  ])
+
+  return { name, symbol, decimals }
+}
+
+/** Chain Generation **/
+
+async function main() {
+  console.log('Running chain generation...')
+  const eta = new Eta({
+    views: './scripts/templates',
+    debug: true,
+    autoTrim: [false, false],
+  })
+
+  const mainnetHttp = process.env.MAINNET_RPC_URL
+    ? [process.env.MAINNET_RPC_URL]
+    : mainnet.rpcUrls.default.http
+  const mainnetClient = createPublicClient({
+    chain: { ...mainnet, rpcUrls: { default: { http: mainnetHttp } } },
+    transport: http(),
+  })
+
+  const sepoliaHttp = process.env.SEPOLIA_RPC_URL
+    ? [process.env.SEPOLIA_RPC_URL]
+    : sepolia.rpcUrls.default.http
+  const sepoliaClient = createPublicClient({
+    chain: { ...sepolia, rpcUrls: { default: { http: sepoliaHttp } } },
+    transport: http(),
+  })
+
+  for (const network of NETWORKS) {
+    console.log(`Generating ${network}`)
+    const client = network === 'mainnet' ? mainnetClient : sepoliaClient
+
+    const configPath = path.join(
+      SUPERCHAIN_REGISTRY_PATH,
+      'superchain',
+      'configs',
+      network,
+    )
+    const entries = fs
+      .readdirSync(configPath)
+      .filter((entry) => !entry.includes('superchain'))
+    const chainDefs = await Promise.all(
+      entries.map(async (entry) => {
+        const chainConfig = toml.parse(
+          fs.readFileSync(`${configPath}/${entry}`, 'utf8'),
+        )
+
+        const addresses = chainConfig.addresses as Record<string, Address>
+        const l1Addresses = {
+          // Referenced as `portal` in viem
+          portal: addresses.OptimismPortalProxy,
+
+          // Standard Deployments
+          l1StandardBridge: addresses.L1StandardBridgeProxy,
+          l1Erc721Bridge: addresses.L1ERC721BridgeProxy,
+          l1CrossDomainMessenger: addresses.L1CrossDomainMessengerProxy,
+          systemConfig: addresses.SystemConfigProxy,
+        }
+
+        if (addresses.DisputeGameFactoryProxy) {
+          l1Addresses['disputeGameFactory'] = addresses.DisputeGameFactoryProxy
+        }
+        if (addresses.L2OutputOracleProxy) {
+          l1Addresses['l2OutputOracle'] = addresses.L2OutputOracleProxy
+        }
+
+        const normalizedName = entry
+          .replace('.toml', '')
+          .replace('-testnet', '')
```

Comment:
if you dont want to rely on the file name for this, there is also the `identifier` field on chainList.json: https://github.com/ethereum-optimism/superchain-registry/blob/e2d3490729b20a649281899c2c286e6e12db57f3/chainList.json#L59. That could potentially be more stable.

Reply:
ahh yea this is a good callout. Need to be able to map this identifier to the filename in order to parse the address list